fastestimator/backend/dice_score.py

```diff
@@ -0,0 +1,148 @@
+               channel_average: bool = False,
+               epsilon: float = 1e-6) -> Tensor:
+    """
+    Compute Dice score.
+
+    This method can be used with Numpy data:
+    ```python
+    true = np.array([[[[0, 1, 1], [1, 0, 1], [1, 0, 1]], [[0, 1, 1], [1, 0, 1], [1, 0, 1]], [[0, 1, 1], [1, 0, 1], [1, 0, 1]]]])
+    pred = np.array([[[[0, 1, 0], [1, 0, 0], [1, 0, 1]], [[0, 1, 1], [1, 0, 1], [0, 0, 0]], [[0, 0, 1], [1, 0, 1], [1, 0, 1]]]])
+    b = fe.backend.dice_score(y_pred=pred, y_true=true)  # 0.161
+    b = fe.backend.dice_score(y_pred=pred, y_true=true, soft_dice=True)  # 0.161
+    b = fe.backend.dice_score(y_pred=pred, y_true=true, channel_average=True)  # 0.1636
+    b = fe.backend.dice_score(y_pred=pred, y_true=true, sample_average=True)  # 0.161
+
+    This method can be used with TensorFlow tensors:
+    ```python
+    true = tf.constant([[[[0, 1, 1], [1, 0, 1], [1, 0, 1]], [[0, 1, 1], [1, 0, 1], [1, 0, 1]], [[0, 1, 1], [1, 0, 1], [1, 0, 1]]]])
+    pred = tf.constant([[[[0, 1, 0], [1, 0, 0], [1, 0, 1]], [[0, 1, 1], [1, 0, 1], [0, 0, 0]], [[0, 0, 1], [1, 0, 1], [1, 0, 1]]]])
+    b = fe.backend.dice_score(y_pred=pred, y_true=true)  # 0.161
+    b = fe.backend.dice_score(y_pred=pred, y_true=true, soft_dice=True)  # 0.161
+    b = fe.backend.dice_score(y_pred=pred, y_true=true, channel_average=True)  # 0.1636
+    b = fe.backend.dice_score(y_pred=pred, y_true=true, sample_average=True)  # 0.161
+    ```
+
+    This method can be used with PyTorch tensors:
+    ```python
+    true = torch.tensor([[[[0, 1, 1], [1, 0, 1], [1, 0, 1]], [[0, 1, 1], [1, 0, 1], [1, 0, 1]], [[0, 1, 1], [1, 0, 1], [1, 0, 1]]]])
+    pred = torch.tensor([[[[0, 1, 0], [1, 0, 0], [1, 0, 1]], [[0, 1, 1], [1, 0, 1], [0, 0, 0]], [[0, 0, 1], [1, 0, 1], [1, 0, 1]]]])
+    b = fe.backend.dice_score(y_pred=pred, y_true=true)  # 0.161
+    b = fe.backend.dice_score(y_pred=pred, y_true=true, soft_dice=True)  # 0.161
+    b = fe.backend.dice_score(y_pred=pred, y_true=true, channel_average=True)  # 0.1636
+    b = fe.backend.dice_score(y_pred=pred, y_true=true, sample_average=True)  # 0.161
+    ```
+
+    ```
+    Args:
+        y_pred: Prediction with a shape like (Batch, C, H, W) for torch and (Batch, H, W, C) for tensorflow or numpy. dtype: float32 or float16.
+        y_true: Ground truth class labels with a shape like `y_pred`. dtype: int or float32 or float16.
+        soft_dice: Whether to square elements. If True, square of elements is added.
+        sample_average: Whether to average the element-wise dice score.
+        channel_average: Whether to average the channel wise dice score.
+
+    Returns:
+        The dice score between `y_pred` and `y_true`. A scalar if `average_sample` is True, else a
+        tensor with the shape (Batch).
+
+    Raises:
+        AssertionError: If `y_true` or `y_pred` are unacceptable data types. if data type is other than np.array, tensor.Tensor, tf.Tensor.
+    """
+    y_true = convert_input_precision(y_true)
```

Comment:
converting precision here isn't necessary. 

cast y_true and epsilon to be the same type as y_red